design-proposals/edge-infra-cve-tracking.md

```diff
@@ -0,0 +1,184 @@
+# Design Proposal: CVE Tracking for OS Profile and Edge Nodes
+
+Author(s): Daniele Moro
+
+Last updated: 2025-05-08
+
+## Abstract
+
+This proposal describes a mechanism for tracking CVEs (Common Vulnerabilities
+and Exposures) for OS Profiles and Edge Nodes in the Edge Infrastructure Manager
+(EIM).
+
+## Proposal
+
+Currently, we have no mechanisms to track CVEs related to the OS in Edge
+Infrastructure Manager. Ideally, we should have a mechanism to track that a
+certain version of the OS is affected by a certain CVE, and that a set of Edge
+Nodes are affected too. Also, the user of EIM should be aware of this, and it
+should be provided with information about which version of the OS fixed the
+given CVE. Users of EIM should be also able to see both forward and backward
+list of fixed CVEs. In order to understand the effect on CVEs on updates.
+
+Requirements:
+- Report existing CVEs for a given OS Profile version
+- Report fixed CVEs on a given OS Profile version
+- Report list of CVEs that affect an Instance at runtime
+- Do backward and forward fixed CVEs tracking between OSProfile versions
+
+The proposal is to extend both the OS Profile and the Instance resources to
+contain information about CVEs.
+
+**OS Profile**:
+- `existing_cves`: a list of CVEs that affect the given OS Profile version. This
+  is filled from information provided by the OS vendor.
+- `fixed_cves`: a list of CVEs that was fixed in the given OS Profile Version.
+  This is filled from information provided by the OS vendor.
+
+**Instance**:
+- `cves`: a list of CVEs that affects the given Instance at runtime. The
```

Comment:
maybe for consistency we call them `existing_cves` here as well ? I'm pondering what is the best approach.